src/components/YieldPools/ProMMFarms.tsx

```diff
@@ -116,6 +145,42 @@ function ProMMFarms({ active }: { active: boolean }) {
     setSeletedPoolId(null)
   }
 
+  const renderAnnouncement = () => {
+    if (activeTab === 'active' && isSomeAffectedFarmsActive && isUserDepositedIntoPreviousFarms) {
+      if (isSomeUserDepositedNFTsFailed) {
+        return <AnnouncementForAffectedUsers />
+      }
+
+      return <AnnouncementForNotYetAffectedUsers />
+    }
+
+    if (activeTab === 'ended' && isSomeAffectedFarmsEnded && isUserDepositedIntoPreviousFarms) {
+      return <AnnouncementForAffectedUsers />
+    }
+
+    return null
+  }
+
+  const shouldShowFarmVersionSelector = isSomeAffectedFarmsActive && isUserDepositedIntoPreviousFarms
+  useEffect(() => {
+    setSelectedFarmVersion(version => {
+      if (activeTab !== 'active') {
+        return undefined
+      }
+
+      // select farm version when it's not set yet
+      if (shouldShowFarmVersionSelector && !version) {
+        return FarmVersion.New
+      }
+
+      if (!shouldShowFarmVersionSelector) {
```

Comment:
we can combine condition to shoter code. 
```
if (activeTab !== 'active' || !shouldShowFarmVersionSelector) {
   return undefined
}
```

Reply:
Nice catch. Fixed